.github/actions/generate-coverage/tests/test_scripts.py

```
@@ -121,24 +117,47 @@ def __init__(self) -> None:
             self.killed = False
             self.waited = False
 
+        def poll(self) -> None:
+            """Report that the process is still running."""
+
         def kill(self) -> None:
+            """Record that the process was killed when lifecycle tracking is enabled."""
             if track_lifecycle:
                 self.killed = True
 
         def wait(self, timeout: float | None = None) -> int:
```

Ruff (0.15.10)

[warning] Unused method argument: `timeout`

(ARG002)